torch/distributed/_tensor/ops/tensor_ops.py

```diff
@@ -598,3 +598,71 @@ def _update_schema_suggestion_for_cat(
         )
     ]
     return output_sharding
+
+
+@register_prop_rule(aten.split.Tensor)
+def split_rule(op_schema: OpSchema) -> OutputSharding:
+    output_spec_list: List[DTensorSpec] = []
+    input_spec = cast(DTensorSpec, op_schema.args_schema[0])
+    ndim = input_spec.ndim
+    split_size_or_sections = op_schema.args_schema[1]
+    dim = 0
+    if len(op_schema.args_schema) > 2:
+        dim = cast(int, op_schema.args_schema[2])
+    dim = normalize_dim(dim, ndim)
+
+    # TODO: just like slice op, split replicates before splitting
+    # on a sharded dimension
+    # TODO: shall we consider partial???
+    # TODO: consider splitting an empty tensor???
+    need_reshard = False
+    if is_tensor_dim_sharded(input_spec, dim=dim):
+        need_reshard = True
+        input_spec = DTensorSpec(
+            mesh=input_spec.mesh,
+            placements=unshard_tensor_dim(input_spec.placements, dim=dim),
+            shape=input_spec.shape,
+            ndim=input_spec.ndim,
+        )
```

Comment:
nit: add a check to partial input_spec and raise NotImplementedError so we know to implement this later?

Reply:
sounds good!